examples/sample-cmd/main.go

```diff
@@ -23,6 +25,31 @@ func main() {
 		return fmt.Sprintf("Hello %s!", c.Param("name")), nil
 	})
 
+	app.SubCommand("spinner", func(ctx *gofr.Context) (interface{}, error) {
+		// intialize the spinner and defer stop it
+		defer terminal.NewDotSpinner(ctx.Out).Spin().Stop()
+
+		// stimulate a time taking process
+		time.Sleep(2 * time.Second)
+
+		return "Process Complete", nil
+	})
+
+	app.SubCommand("progress", func(ctx *gofr.Context) (interface{}, error) {
+		p := terminal.NewProgressBar(ctx.Out, 100)
+
+		for i := 1; i <= 100; i++ {
+			// do a time taking process or compute a small subset of a bigger problem,
+			// this could be processing batches of a data set.
+			time.Sleep(50 * time.Millisecond)
+
+			// increment the progress to display on the progress bar.
+			p.Incr(int64(1))
+		}
```

Comment:
This sub command won't expire if the context is canceled

I would use a check of ctx.Err() to get out the loop